src/compiler/emitter.ts

```diff
@@ -2709,6 +2709,17 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                     }
                     else {
                         if (node.flags & NodeFlags.Default) {
+                            if (compilerOptions.module === ModuleKind.CommonJS || compilerOptions.module === ModuleKind.AMD || compilerOptions.module === ModuleKind.UMD) {
+                                write("Object.defineProperty(exports, \"__esModule\", {");

```

Comment:
If you want to emulate this automatically, putting the define property behind a `languageVersion >= ScriptTarget.ES5` and adding a `write("exports.__esModule = true");` for es3 would work.